### PR TITLE
Add simple unit test + dependencies

### PR DESCRIPTION
**Pull request by Hakem22**

### CODE REVIEW

File: src/room-creation/services/room-service.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import type { Room } from "../models/room";
+import type { RoomCreationRequest } from "../models/room-creation-request";
+
+export async function create(request: RoomCreationRequest) {
```
Comment (DanielMuckelbauer) [inline src/room-creation/services/room-service.ts:4]: As discussed, I think we should reflect in the filename that this is not a service, but rather only a function. Maybe just call it `createRoom.ts`? 
Whatever you choose to name it, don't forget to also change it in the test and the `describe` please :)

File: src/room-creation/room-creation-modal.test.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        const component = render(RoomCreationModal);
+    })
+    
+     test("renders the dialog when the user clicks on open dialog button", async () => {
+         const component = render(RoomCreationModal);
+         const openModal = component.getAllByText('Create a Room')
+
+         await fireEvent.click(openModal[0]);
+
+         expect(component.getByText("Room Creation")).toBeTruthy();
```
Comment (DanielMuckelbauer) [inline src/room-creation/room-creation-modal.test.ts:15]: I kind of like this api to test components, very nice :)

File: src/room-creation/room-creation-modal.test.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { fireEvent, render } from "@testing-library/svelte";
+import RoomCreationModal from "./RoomCreationModal.svelte";
```
Comment (DanielMuckelbauer) [inline src/room-creation/room-creation-modal.test.ts:2]: I have no idea about svelte of course, but I think we should probably also embrace the PascalCase naming convention of the `.svelte` components for their test file names, so maybe `RoomCreationModalTests`? What is your opinion? I am afraid we are too influenced by the Angular naming :)

File: src/room-creation/RoomCreationModal.svelte [hunk 4]
```diff
@@ -3,44 +3,36 @@
     import { Row, Col, TextField } from 'svelte-materialify';
     import { GameType } from "./models/game-type";
     import { Room } from "./models/room"
+    import { create } from "./services/room-service";
+    import type { RoomCreationRequest } from "./models/room-creation-request";
 
     export let roomForm = <Room>{};
     const items = Object.values(GameType).filter(value => isNaN(Number(value)));
-    let active = false
+    export let active = false
```
Comment (DanielMuckelbauer) [inline src/room-creation/RoomCreationModal.svelte:12]: Could you explain why this change is necessary? It was my understanding that this syntax creates a property that can be set by the parent, yet I don't see where this is set.

File: svelte.config.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const sveltePreprocess = require("svelte-preprocess");
+
+module.exports = {
+    preprocess: sveltePreprocess(),
+};
```
Comment (DanielMuckelbauer) [inline svelte.config.js:5]: Maybe you can explain the meaning of this either here or in our next call :) Should not block the PR, just curious

File: src/room-creation/services/room-service.ts (hunk 1 above)
Comment (Hakem22) [reply src/room-creation/services/room-service.ts:4]: Hmm, my intentions were to add the other HTTP calls that are more or less relation to the room to the same file. but I am also fine with having them in separate files and thus rename this service

File: src/room-creation/RoomCreationModal.svelte (hunk 4 above)
Comment (Hakem22) [reply src/room-creation/RoomCreationModal.svelte:12]: I added cos I wanted to test something, and forgot to remove it. it will be done!

File: src/room-creation/room-creation-modal.test.ts (hunk 3 above)
Comment (Hakem22) [reply src/room-creation/room-creation-modal.test.ts:2]: I am also unhappy, about the fact that I had to use PascalCase and snake-case in same project is ugly :) I was waiting for this comment.
1- Either we keep all Ts files with snake-case.
2- we change all Ts files to PascalCase including `model.ts`, `service.ts`, and `test.ts` files?

I don't like the idea of having only `test.ts` in PascalCase, what do you think?

File: src/room-creation/services/room-service.ts (hunk 1 above)
Comment (DanielMuckelbauer) [reply src/room-creation/services/room-service.ts:4]: I see, that makes more sense. I just somehow take offense of it being called service, since it is not really. 
I have no better suggestion for now though to be honest, so don't put to much effort into it. I will have to touch/change it soon anyway, and depending on how it looks like then we can decide on a fitting name.

File: src/room-creation/room-creation-modal.test.ts (hunk 3 above)
Comment (DanielMuckelbauer) [reply src/room-creation/room-creation-modal.test.ts:2]: As far as I can tell, PascalCase is idiomatic for svelte. I am fine with both conventions, as long as it is consistent. What do you prefer?

File: svelte.config.js (hunk 5 above)
Comment (Hakem22) [reply svelte.config.js]: The svelte-preprocess is used to enable the use of typescript, this configuration has been already added to rollup.config.js line 43. That file was added when I was testing the firebase deploy and again forgot to remove it. Bottom line sorry for the mess and thank you for spotting those stuff :)

File: src/room-creation/room-creation-modal.test.ts (hunk 3 above)
Comment (Hakem22) [reply src/room-creation/room-creation-modal.test.ts:2]: I have moved the test files to a separate folder, I have kept the snake-case for typescript files for now until we find a convincing answer to the perfect naming convention that we should follow. then we can add it to lint rules as well. sorry for now :(

File: src/room-creation/services/room-service.ts (hunk 1 above)
Comment (Hakem22) [reply src/room-creation/services/room-service.ts:4]: I think it will be changed, then let's wait on how it gonna progress then we cand decide for the name :)

Comment (DanielMuckelbauer) [inline src/room-creation/services/room-service.ts:4]: The return type is missing, but this function will be obsolete soon anyway so don't bother :)

File: src/room-creation/RoomCreationModal.svelte [hunk 6]
```diff
@@ -11,36 +13,26 @@
...
-        }
+        const roomCreationRequest = <RoomCreationRequest>roomForm
+        roomForm = await create(roomCreationRequest);
     }
 </script>
 
 <MaterialApp>
     <div class="text-center">
-        <Button on:click={openModal}>Create a Room</Button>
+        <Button id="modal-button" on:click={openModal}>Create a Room</Button>
```
Comment (DanielMuckelbauer) [inline src/room-creation/RoomCreationModal.svelte:30]: What is this used for? I don't see it :(

File: src/room-creation/services/room-service.ts (hunk 1 above)
Comment (Hakem22) [reply src/room-creation/services/room-service.ts]: Added

File: src/room-creation/RoomCreationModal.svelte (hunk 6 above)
Comment (Hakem22) [reply src/room-creation/RoomCreationModal.svelte]: Removed

**Review by DanielMuckelbauer — APPROVED** [review]
